src/Module/Diaspora/Receive.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Friendica\Module\Diaspora;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Core\Config\Configuration;
+use Friendica\Model\User;
+use Friendica\Network\HTTPException;
+use Friendica\Protocol\Diaspora;
+use Friendica\Util\Network;
+use Psr\Log\LoggerInterface;
+
+/**
+ * This module is part of the Diaspora protocol.
+ * It is used for receiving single posts either for public or for a specific user.
+ */
+class Receive extends BaseModule
+{
+	/** @var LoggerInterface */
+	private static $logger;
+
+	public static function init()
+	{
+		/** @var LoggerInterface $logger */
+		self::$logger = self::getClass(LoggerInterface::class);
+	}
+
+	public static function post()
+	{
+		/** @var Configuration $config */
+		$config = self::getClass(Configuration::class);
+
+		$enabled = $config->get('system', 'diaspora_enabled', false);
+		if (!$enabled) {
+			self::$logger->info('Diaspora disabled.');
+			throw new HTTPException\InternalServerErrorException('Diaspora disabled.');
```

Comment:
Ah damn I didn't see this one that should have been "ForbiddenException" instead.

Reply:
Fixed with #7770